File: lecture2/pset2/readability/readability.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>

float avarage_per_100(float number, float word);

int main(void)
{
    string text = get_string("Text: ");

    int letter = 0;
    int word = 1;
    int sentece = 0;
    int i = 0;
    float result;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z' || text[i] >= 'A' && text[i] <= 'Z')
        {
            letter++;
        }
        else if(text[i] == ' ')
        {
            word++;
        }
        else if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentece++;
        }
    }

    result = 0.0588 * avarage_per_100(letter, word) - 0.296 * avarage_per_100(sentece, word) - 15.8;

    if ((int) round(result) > 16)
    {
        printf("Grade 16+\n");
    }
    else if ((int) round(result) < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",(int) round(result));
    }

}

float avarage_per_100(float number, float word)
{
    float result;
    result = number / word * 100;
    return result;
}

File: lecture6/pset6/sentimental-readability/readability.py
from cs50 import get_string
import re


def main():
    text = get_text()
    letters = how_many_letters(text)
    sentences = how_many_sentences(text)
    words = how_many_word(text)

    total = coleman_liau(letters, sentences, words)
    total = round(total)

    if total > 16:
        print("Grade 16+")
    elif total < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {total}")


def coleman_liau(letter, sentence, word):
    letter_avarage = avarage(letter, word)
    sentence_avarage = avarage(sentence, word)
    return 0.0588 * letter_avarage - 0.296 * sentence_avarage - 15.8


def how_many_word(text):
    pattern = '[a-z]+'
    text = text.replace("'", "")
    text = text.replace("-", "")
    text = re.findall(pattern, text, re.I)
    return len(text)


def avarage(letter_or_sentence, words_number):
    return letter_or_sentence / words_number * 100


def how_many_sentences(text):
    pattern = '[!.?]'
    text = re.findall(pattern, text)
    text = ''.join(text)
    return len(text)


def how_many_letters(text):
    pattern = '[a-z]+'
    text = re.findall(pattern, text, re.I)
    text = ''.join(text)
    return len(text)


def get_text():
    text = get_string("Text: ")
    return text


main()
